platform/core.startup/src/org/netbeans/core/startup/Splash.java

```diff
@@ -518,7 +520,7 @@ public void run() {
                         return;
                     }
 
-                    if (statusBox.fm == null) {
+                    if (statusBox == null || statusBox.fm == null) {
```

Comment:
It was never the intention for statusBox to be able to be null here. I failed to see the "if (maxSteps > 0)" condition in SplashPainter.init(). It seems to be an attempt to avoid running init() twice, but maxSteps can be increased in other ways, so this is not quite correct.

I'd rather change "if (maxSteps > 0) { return; }" to "if (statusBox != null) { return; }" in SplashPainter.init(). Then you can get rid of this other null check. Would you mind seeing if this solves the problem?